### ISSUE TITLE
Typescript compiler reports TS1385 error on Adjust class

### ISSUE THREAD

**Issue by ahtokca** [adjust/react_native_sdk#134]
Dear Adjust,

I use tsc for my react native app and I get the error below running `tsc --noemit` check.

```
react-native-adjust/index.d.ts:184:64 - error TS1385: Function type notation must be parenthesized when used in a union type.

```
it doesn't like the line below

```
trackAdRevenue: (source: string, payload: string) => void | (source: AdjustAdRevenue) => void
```

Would it be possible to fix it?

**Comment by uerceg** [comment]
Hi @ahtokca 

Thanks for reporting. We'll take a look at this on our end and hopefully come up with a fix soon. Will keep you posted.

Cheers

**Comment by uerceg** [comment]
We have just released [v4.29.1](https://github.com/adjust/react_native_sdk/releases/tag/v4.29.1) update which should hopefully address this issue. @ahtokca if you can give it a shot and let us know if the issue is still out there or gone, would be amazing. Thanks in advance.

**Comment by uerceg** [comment]
@ahtokca @leeuwd Did you guys have a chance to give this version a shot to see if it fixes the issue?

**Comment by ahtokca** [comment]
@uerceg works like a charm. Thanks a lot! I didn't expect such lightning fast fix :)

**Comment by uerceg** [comment]
Yay. Thanks for getting back to us. Cheers!

**Comment by smacgregor** [comment]
@ahtokca - I get this tsc error now - with the corrected declaration for trackAdRevenue, I now get the following tsc error when passing a string into trackAdRevenue. I'm wondering if you see this too:

Argument of type 'string' is not assignable to parameter of type 'string & AdjustAdRevenue'.
  Type 'string' is not assignable to type 'AdjustAdRevenue'.

Calling code: 

```
  trackAdImpression(source: string, payload: string) {
    Adjust.trackAdRevenue(source, payload);
  }
```

**Comment by uerceg** [comment]
@smacgregor Any better with just released [v4.29.3](https://github.com/adjust/react_native_sdk/releases/tag/v4.29.3)?
